thirdparty/github.go

```diff
@@ -271,16 +296,40 @@ func GetGithubCommits(ctx context.Context, oauthToken, owner, repo, ref string,
 	commits, resp, err := client.Repositories.ListCommits(ctx, owner, repo, &options)
 	if resp != nil {
 		defer resp.Body.Close()
+		if err == nil {
```

Comment:
It's more idiomatic in Go to simplify this (and other functions in this file that do a similar thing) down to fewer nested blocks down to:
```go
if resp != nil { defer resp.Body.Close() }
if err != nil {
    return <the expected result>, nil
}
```
I believe this code is still correct, but it's likelier to cause bugs if anyone changes the logic later on compared to having fewer nested blocks.

Reply:
I only saw this review after submitting mine, so there is some overlap 🙈

Reply:
discussed!